extensions/resteasy-jsonb/runtime/src/main/java/io/quarkus/resteasy/jsonb/runtime/QuarkusJsonbContextResolver.java

```diff
@@ -0,0 +1,24 @@
+package io.quarkus.resteasy.jsonb.runtime;
+
+import javax.annotation.Priority;
+import javax.json.bind.Jsonb;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.ext.ContextResolver;
+import javax.ws.rs.ext.Provider;
+
+import io.quarkus.arc.Arc;
+
+@Provider
+@Priority(Priorities.USER + 10) // give it a priority that ensures that user supplied ContextResolver classes override this one
+public class QuarkusJsonbContextResolver implements ContextResolver<Jsonb> {
+
+    private Jsonb jsonb;
+
+    @Override
+    public Jsonb getContext(Class<?> type) {
+        if (null == jsonb) {
```

Comment:
I think this is not thread-safe, a begning race condition can occurs here.
Maybe getting the bean from Arc at postConstruct will be better ?

Reply:
I don't think it really matters to be honest. The returned object will always be the same, so even if multiple threads access it, at works they will both go to Arc and return the same object.

Reply:
I'm not totally sure it will be safe. Standard injection does not work in `ContextResolver`? I'm wondering if you could just inject the instance via `@Inject`?

Reply:
That was the first I tried and it didn't work

Reply:
But let me give it another shot

Reply:
It doesn't work. Do we have some way of making providers beans?

Reply:
I'll dig into this some more and see if we can do something

Reply:
I thought they were to be honest.

Another option would be to initialize it in the constructor if it's not too early.

Reply:
It shouldn't be too early. The only reason I wanted to do it was to avoid any unecessary initialization at startup.

If I can't figure out why the Providers aren't actually injecting (which TBH before doing this excersise I also thought were capable of injection), that's what I'll end up doing

Reply:
I'm surprised that using `@PostConstruct` didn't works. It should allow to add init blocks after the bean factory was initalized ... maybe a bug in Arc ?

Reply:
So now everything should work as expected. I had missed adding the providers as `AdditionalBean`s